### PR TITLE
Add/Edit/View Bonds

### PR DESCRIPTION
**Pull request by mmumby**
FE work for securities and bonds
Work completed: 
- POST bond
-GET bonds
-PUT bond
- release and confiscate bonds, if confiscated the bond_type gets converted to cash
- Calculated amount held by mine, and amount held and confiscated by permit
- added "securities_total" field to permit amendment put

Screen Shot 2020-03-23 at 1 49 05 PM
Screen Shot 2020-03-23 at 1 49 14 PM
Screen Shot 2020-03-23 at 1 49 24 PM

### CODE REVIEW

File: services/core-web/src/components/mine/Securities/MineBondTable.js [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+      key: "permit_no",
+      render: (text) => <div title="Permit No.">{text}</div>,
+    },
+    {
+      title: "Total Assessed",
+      dataIndex: "security_total",
+      key: "security_total",
+      render: (text, record) => (
+        <div title="Total Assessed">
+          {formatMoney(record.permit_amendments[0].security_total) || Strings.EMPTY_FIELD}
```
[the remark this answers is not in the document]
Comment (mmumby) [reply services/core-web/src/components/mine/Securities/MineBondTable.js]: yes they do, upon creation of a permit an "initial amendment" is created

File: services/core-web/src/components/modalContent/ViewBondModal.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  statusTypeOptionsHash: PropTypes.objectOf(PropTypes.string).isRequired,
+  bond: CustomPropTypes.bond.isRequired,
+};
+
+export const ViewBondModal = (props) => {
+  const address = {
+    address_line_1: props.bond.institution_street,
+    city: props.bond.institution_city,
+    sub_division_code: props.bond.institution_province,
+    post_code: props.bond.institution_postal_code,
```
[the remark this answers is not in the document]
Comment (mmumby) [reply services/core-web/src/components/modalContent/ViewBondModal.js]: discussed on video.

File: services/core-web/src/components/Forms/Securities/BondForm.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+          />
+        </Form.Item>
+      </Col>
+      <Col md={12} xs={24}>
+        <Form.Item>
+          <Field
+            id="institution_postal_code"
+            name="institution_postal_code"
+            label="Postal Code"
+            placeholder="e.g xxxxxx"
```
[the remark this answers is not in the document]
Comment (mmumby) [reply services/core-web/src/components/Forms/Securities/BondForm.js]: discussed on video.

File: services/core-web/common/utils/helpers.js [hunk 4]
```diff
@@ -259,3 +259,8 @@ export const flattenObject = (ob) => {
   }
   return toReturn;
 };
+
+export const formatMoney = (value) => {
+  const number = Number(value);
+  return number.toLocaleString("en-US", { style: "currency", currency: "USD" });
```
[the remark this answers is not in the document]
Comment (mmumby) [reply services/core-web/common/utils/helpers.js]: good call, completely overlooked that. fixed.

File: services/core-web/src/components/mine/Securities/MineBondTable.js [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+              </Dropdown>
+            </AuthorizationWrapper>
+          </div>
+        );
+      },
+    },
+  ];
+
+  const bondsByPermit = (permit) =>
+    props.bonds.filter(({ permit_guid }) => permit_guid === permit.permit_guid);
```
[the remark this answers is not in the document]
Comment (mmumby) [reply services/core-web/src/components/mine/Securities/MineBondTable.js]: I remember why I didn't move these to selectors, I don't have reference to the permit and selectors do not accept params. In this situation, I needed to pass the row permit into this function to get the total.

File: services/core-web/src/components/mine/Securities/MineBondTable.js [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+  const activeBondCount = (permit) =>
+    props.bonds.filter(
+      ({ permit_guid, bond_status_code }) =>
+        permit_guid === permit.permit_guid && bond_status_code === "ACT"
+    ).length;
+  const getSum = (status, permit) =>
```
[the remark this answers is not in the document]
Comment (mmumby) [reply services/core-web/src/components/mine/Securities/MineBondTable.js]: same as above, no reference to the row permit if done in the selector.

File: services/core-web/common/utils/helpers.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (mmumby) [reply services/core-web/common/utils/helpers.js]: Screen Shot 2020-03-24 at 3 42 02 PM


It prepends with CA which is not the desired behaviour, I have switched it back to the default

File: services/core-web/src/components/mine/Securities/MineBondTable.js [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+      sorter: (a, b) => (a.bond_type_code > b.bond_type_code ? -1 : 1),
+      render: (text) => (
+        <div title="Type">{props.bondTypeOptionsHash[text] || Strings.EMPTY_FIELD}</div>
+      ),
+    },
+    {
+      title: "Amount",
+      dataIndex: "amount",
+      key: "amount",
+      render: (text) => <div title="Amount">{formatMoney(text) || Strings.EMPTY_FIELD}</div>,
```
[the remark this answers is not in the document]
Comment (mmumby) [reply services/core-web/src/components/mine/Securities/MineBondTable.js]: updated, it shows 'N/A' for null values, $0.00 for everything else

File: services/core-web/common/utils/helpers.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (exzizt) [reply services/core-web/common/utils/helpers.js]: There are [additional params](https://www.techonthenet.com/js/number_tolocalestring.php) to configure it, but no problem.

**Pull request merged**